api/helpers.go

```diff
@@ -107,3 +110,99 @@ func (a *API) getReferrer(r *http.Request) string {
 	}
 	return referrer
 }
+
+var privateIPBlocks []*net.IPNet
+
+func init() {
+	for _, cidr := range []string{
+		"127.0.0.0/8",    // IPv4 loopback
+		"10.0.0.0/8",     // RFC1918
+		"100.64.0.0/10",  // RFC6598
+		"172.16.0.0/12",  // RFC1918
+		"192.0.0.0/24",   // RFC6890
+		"192.168.0.0/16", // RFC1918
+		"169.254.0.0/16", // RFC3927
+		"::1/128",        // IPv6 loopback
+		"fe80::/10",      // IPv6 link-local
+		"fc00::/7",       // IPv6 unique local addr
+	} {
+		_, block, _ := net.ParseCIDR(cidr)
+		privateIPBlocks = append(privateIPBlocks, block)
+	}
+}
+
+func isPrivateIP(ip net.IP) bool {
+	for _, block := range privateIPBlocks {
+		if block.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func removeLocalhostFromPrivateIPBlock() *net.IPNet {
+	var localhost *net.IPNet
+	localhost, privateIPBlocks = privateIPBlocks[0], privateIPBlocks[1:]
```

Comment:
we should not rely on localhost being the first entry...

Reply:
Yeah, though about it but was lazy 🙈 
Will update!